src/components/decimal/decimal.component.js

```diff
@@ -11,247 +16,261 @@ import LocaleContext from "../../__internal__/i18n-context";
+          value: {
+            formattedValue: formatValue(toStandardDecimal(formatted)),
+            rawValue: raw || toStandardDecimal(formatted),
+          },
+        },
+      };
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    []
+  );
+
+  /**
+   * Determine if the component is controlled at the time of call
+   */
+  function isControlled(arg) {
+    return arg !== undefined;
+  }
+
+  const handleOnChange = (ev) => {
+    const val = ev.target.value;
+    setStateValue(val);
+    if (onChange) onChange(createEvent(val));
+  };
+
+  const handleOnBlur = (ev) => {
+    const { value: updatedValue } = ev.target;
+    let event;
+
+    if (updatedValue) {
+      const standardVisible = toStandardDecimal(updatedValue);
+      const formattedValue = isNaN(standardVisible)
+        ? updatedValue
+        : formatValue(standardVisible);
+
+      event = createEvent(formattedValue, standardVisible);
+      setStateValue(formattedValue);
+    } else {
+      event = createEvent(emptyValue);
+      setStateValue(emptyValue);
     }
 
-    const withoutDelimiters = this.removeDelimiters(valueWithoutNBS);
-    return withoutDelimiters.replace(new RegExp(`\\${separator}`, "g"), ".");
+    if (onBlur) onBlur(event);
   };
 
-  getSeparator(separatorType) {
-    const numberWithGroupAndDecimalSeparator = 10000.1;
-    return Intl.NumberFormat(this.props.locale || this.context.locale())
-      .formatToParts(numberWithGroupAndDecimalSeparator)
-      .find((part) => part.type === separatorType).value;
-  }
+  const isComponentControlled = isControlled(value);
 
-  render() {
-    const { name, defaultValue, locale, ...rest } = this.props;
-    return (
-      <>
-        <Textbox
-          {...rest}
-          onChange={this.onChange}
-          onBlur={this.onBlur}
-          value={this.state.visibleValue}
-        />
-        <input
-          name={name}
-          value={this.toStandardDecimal(this.state.visibleValue)}
-          type="hidden"
-          data-component="hidden-input"
-        />
-      </>
-    );
-  }
-}
+  useEffect(() => {
+    const unformattedValue = toStandardDecimal(stateValue);
+
+    if (isComponentControlled) {
+      const valueProp = getSafeValueProp(value);
+      if (unformattedValue !== valueProp) {
+        setStateValue(formatValue(value));
+      }
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [value]);
+
+  return (
+    <>
+      <Textbox
+        onKeyPress={onKeyPress}
+        align={align}
+        readOnly={readOnly}
+        required={required}
+        inputWidth={inputWidth}
+        onChange={handleOnChange}
+        onBlur={handleOnBlur}
+        value={stateValue}
+        data-component="decimal"
+        {...rest}
+      />
+      <input
+        name={name}
+        value={toStandardDecimal(stateValue)}
+        type="hidden"
+        data-component="hidden-input"
+      />
+    </>
+  );
+};
 
 Decimal.contextType = LocaleContext;
```

Comment:
Don't think we need this anymore

Reply:
You're right. Functional components can't use `contextType` according to the console warning message I hadn't seen previously.